scripts/lib/CIME/XML/grids.py

```diff
@@ -204,6 +204,10 @@ def _read_config_grids_v2(self, name, compset):
                 lname = prefix[component_gridname]
             if model_grid[component_gridname] is not None:
                 lname += model_grid[component_gridname]
+                if component_gridname == 'atm' and atmnlev is not None:
+                    lname += "z" + atmnlev
+                elif component_gridname == 'lnd' and lndnlev is not None:
+                    lname += "z" + lndnlev
```

Comment:
I think we may need some error checking here. I tried an A compset with a grid alias of ne30z32_ne30z14 and ended up with LND_GRID="nullz14" in env_build.xml
I'm not sure this damages anything but I don't think this is intended.
Maybe modifying the land levmatch:
`levmatch = re.match(r"(.*_)([^_]+)z(\d+)(_[^m].*)$", name)`
and add an additional check to each if block above.